clippy_lints/src/methods/range_zip_with_len.rs

```diff
@@ -0,0 +1,36 @@
+use clippy_utils::diagnostics::span_lint;
+use clippy_utils::source::snippet;
+use clippy_utils::{higher, SpanlessEq};
+use clippy_utils::{is_integer_const, is_trait_method};
+use if_chain::if_chain;
+use rustc_hir::{Expr, ExprKind, QPath};
+use rustc_lint::LateContext;
+use rustc_span::sym;
+
+use super::RANGE_ZIP_WITH_LEN;
+
+pub(super) fn check<'tcx>(cx: &LateContext<'tcx>, expr: &'tcx Expr<'_>, recv: &'tcx Expr<'_>, zip_arg: &'tcx Expr<'_>) {
+    if_chain! {
+        if let ExprKind::MethodCall(iter_path, [iter_recv], _) = recv.kind;
+        if iter_path.ident.name == sym::iter;
```

Comment:
NIT: Could this be done in the `methods` module and the `iter_recv` passed into this function?

Reply:
Done.